src/main/java/com/navin/backend/e_market/controller/ProductController.java

```diff
@@ -17,48 +20,66 @@ public class ProductController {
     @Autowired
     private ProductService productService;
 
+    @Autowired
+    private CustomerRepo customerRepo;
+
     @PostMapping
-    public ResponseEntity<Product> createProduct(@RequestBody Product product)
-    {
+    public ResponseEntity<Product> createProduct(@RequestBody Product product) {
         productService.addProduct(product);
         return new ResponseEntity<>(product, HttpStatus.CREATED);
     }
 
     @GetMapping
-    public ResponseEntity<List<Product>> getAll(){
-        return new ResponseEntity<>(productService.getAll(),HttpStatus.OK);
+    public ResponseEntity<Object> getAll(@RequestHeader("Authorization") String token, @RequestHeader("email") String email) {
+
+
+        String secretKey = customerRepo.findByEmail(email).map(c -> c.getSecretKey()).get();
```

Comment:
you can also use method reference instead of lambda expression 

Customer::getSecretKey